### ISSUE TITLE
Request: Can `detect` be a default value for the version setting?

### ISSUE THREAD

**Issue by robwierzbowski** [jsx-eslint/eslint-plugin-react#3556]
### Setup

I'm using the plugin on version 7.32.2 and ESLint on 8.36.0. My app is using Vite.

First off, thank you for providing a config option and docs for the new ESLint config. At first, I set up the config as so:

```js
import { default as reactRecommended } from 'eslint-plugin-react/configs/recommended.js';

const config = [
  // Note, although the eslint-plugin-react docs suggest using a file glob here, I haven't yet
  // found it necessary.
  reactRecommended,
  {
    files: ['**/*.js', '**/*.ts', '**/*.tsx', '**/*.d.ts'],
    languageOptions,
   // ...config continues
```

When linting with this config I get the warning: 

```
Warning: React version not specified in eslint-plugin-react settings. See https://github.com/jsx-eslint/eslint-plugin-react#configuration .
```

To resolve this warning and use the correct rules for my version of React, I have changed my configuration to:

```js
const config = [
  {
    ...reactRecommended,
    settings: {
      react: {
        version: 'detect',
      },
    },
  },
  {
    files: ['**/*.js', '**/*.ts', '**/*.tsx', '**/*.d.ts'],
    languageOptions,
     // ...config continues
```

### Requesst

Is it possible to make `detect` the default value for the React version setting? This would allow us to apply the plugin's recommended settings in a cleaner, easier to maintain one liner.

**Comment by ljharb** [comment]
That would be a breaking change. However, I suggest you use the airbnb config and NOT this plugin's "recommended" config (which we'll likely remove in the next semver-major), which does set it to detect.

**Comment by robwierzbowski** [comment]
Thanks for the quick response @ljharb, appreciate it! That's all very reasonable.
